packages/components/src/border-control/border-control-dropdown/component.tsx

```diff
@@ -0,0 +1,128 @@
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { closeSmall } from '@wordpress/icons';
+
+/**
+ * Internal dependencies
+ */
+import BorderControlStylePicker from '../border-control-style-picker';
+import Button from '../../button';
+import ColorIndicator from '../../color-indicator';
+import ColorPalette from '../../color-palette';
+import Dropdown from '../../dropdown';
+import { HStack } from '../../h-stack';
+import { VStack } from '../../v-stack';
+import { contextConnect, WordPressComponentProps } from '../../ui/context';
+import { useBorderControlDropdown } from './hook';
+import { StyledLabel } from '../../base-control/styles/base-control-styles';
+
+import type { DropdownProps, PopoverProps } from '../types';
+const noop = () => undefined;
+
+const BorderControlDropdown = (
+	props: WordPressComponentProps< DropdownProps, 'div' >,
+	forwardedRef: React.ForwardedRef< any >
+) => {
+	const {
+		__experimentalHasMultipleOrigins,
+		__experimentalIsRenderedInSidebar,
+		border,
+		colors,
+		disableCustomColors,
+		enableAlpha,
+		indicatorClassName,
+		indicatorWrapperClassName,
+		onReset,
+		onColorChange,
+		onStyleChange,
+		popoverClassName,
+		popoverContentClassName,
+		popoverControlsClassName,
+		resetButtonClassName,
+		enableStyle = true,
+		...otherProps
+	} = useBorderControlDropdown( props );
+
+	const { color, style } = border || {};
+
+	const renderToggle = ( { onToggle = noop } ) => (
+		<Button
+			onClick={ onToggle }
+			variant="tertiary"
+			aria-label={ __( 'Open border options' ) }
```

Comment:
Since this button is not just a trigger for the popover but also an indicator for the current settings, I think it would be useful for the aria-label to reflect that information too (i.e. current border color and style). Otherwise you'd have to tab through everything in the popover to understand what the current settings are.

Reply:
I've attempted to improve the aria-label information to reflect color and style selection in https://github.com/WordPress/gutenberg/pull/37769/commits/1782b43632e9829cd6ff847b897bb4cab7169f40.

There are several possible combinations so I'm not sure about the approach I've taken or the wording. I went with the more complete and verbose option, following along similar lines as the [recent updates](https://github.com/WordPress/gutenberg/pull/39386/files#diff-b53300a311404baec5a0c0cfc1d898fa89050f29959680b6317243be716c3691R203) to the custom color button

Reply:
Awesome. We can always tweak it if people find it too verbose.